x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/import_rule_modal/index.tsx

```diff
@@ -0,0 +1,160 @@
+      if (selectedFiles != null) {
+        setIsImporting(true);
+        const reader = new FileReader();
+        reader.onload = async event => {
+          // @ts-ignore type is string, not ArrayBuffer as FileReader.readAsText is called
+          const importedRules = ndjsonToJSON(event?.target?.result ?? '');
+
+          const decodedRules = pipe(
+            RulesSchema.decode(importedRules),
+            fold(errors => {
+              cleanupAndCloseModal();
+              dispatchToaster({
+                type: 'addToaster',
+                toast: {
+                  id: uuid.v4(),
+                  title: i18n.IMPORT_FAILED,
+                  color: 'danger',
+                  iconType: 'alert',
+                  errors: failure(errors),
+                },
+              });
+              throw new Error(failure(errors).join('\n'));
+            }, identity)
+          );
+
+          const duplicatedRules = await duplicateRules({ rules: decodedRules, kbnVersion });
+          importComplete();
+          cleanupAndCloseModal();
+
+          dispatchToaster({
+            type: 'addToaster',
+            toast: {
+              id: uuid.v4(),
+              title: i18n.SUCCESSFULLY_IMPORTED_RULES(duplicatedRules.length),
+              color: 'success',
+              iconType: 'check',
+            },
+          });
+        };
+        Object.values(selectedFiles).map(f => reader.readAsText(f));
+      }
+    }, [selectedFiles]);
+
+    return (
+      <>
+        {showModal && (
+          <EuiOverlayMask>
+            <EuiModal onClose={closeModal} maxWidth={'750px'}>
+              <EuiModalHeader>
+                <EuiModalHeaderTitle>{i18n.IMPORT_RULE}</EuiModalHeaderTitle>
+              </EuiModalHeader>
+
+              <EuiModalBody>
+                <EuiText size="s">
+                  <h4>{i18n.SELECT_RULE}</h4>
+                </EuiText>
+
+                <EuiSpacer size="s" />
+                <EuiFilePicker
+                  id="rule-file-picker"
+                  multiple
+                  initialPromptText={i18n.INITIAL_PROMPT_TEXT}
+                  onChange={(files: FileList) => {
+                    setSelectedFiles(Object.keys(files).length > 0 ? files : null);
+                  }}
+                  display={'large'}
+                  fullWidth={true}
+                  isLoading={isImporting}
+                />
+                <EuiSpacer size="s" />
+                <EuiCheckbox
+                  id="rule-overwrite-saved-object"
+                  label={i18n.OVERWRITE_WITH_SAME_NAME}
+                  disabled={true}
+                  onChange={() => {}}
+                />
+              </EuiModalBody>
+
+              <EuiModalFooter>
+                <EuiButtonEmpty onClick={closeModal}>{i18n.CANCEL_BUTTON}</EuiButtonEmpty>
+                <EuiButton
+                  onClick={importRules}
+                  disabled={selectedFiles == null || isImporting}
```

Comment:
nit: Don't you want your fancy smancy `??`  like so: `selectedFiles == null ?? isImporting`  ;-)